scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1289,13 +1315,21 @@ private void handleTimeout(DTLSFlight flight) {
 			LOGGER.log(Level.FINE, "Re-transmitting flight for [{0}], [{1}] retransmissions left",
 					new Object[]{flight.getPeerAddress(), max - flight.getTries() - 1});
 
-			flight.incrementTries();
-			sendFlight(flight);
+			try {
+				flight.incrementTries();
+				flight.setNewSequenceNumbers();
```

Comment:
Yes, but `DTLSConnector.sendFlight(DTLSFlight)` explicitly checked for `flight.getTries() > 0` and only in that case set new sequence numbers on the flight, i.e. this method explicitly checked whether the flight was a re-transmisson. However, we already have a method that is *only* executed for flights that are re-transmitted and that is `DTLSConnector.handleTimeout()` which is invoked by the `TimerTask` scheduled for the outbound flight ...